src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -1281,8 +1282,8 @@ public static IEnumerable<object[]> CustomInterfaces_Dictionaries()
         [Fact]
         public async Task IReadOnlyDictionary_NotSupportedKey()
         {
-            await Assert.ThrowsAsync<NotSupportedException>(async () => await Serializer.DeserializeWrapper<IReadOnlyDictionary<Uri, int>>(@"{""http://foo"":1}"));
-            await Assert.ThrowsAsync<NotSupportedException>(async () => await Serializer.SerializeWrapper(new GenericIReadOnlyDictionaryWrapper<Uri, int>(new Dictionary<Uri, int> { { new Uri("http://foo"), 1 } })));
+            await Assert.ThrowsAsync<NotSupportedException>(async () => await Serializer.DeserializeWrapper<IReadOnlyDictionary<JsonNode, int>>(@"{""key"":1}"));
```

Comment:
what changed with regard to Uris?

Reply:
They now support property name serialization.